crates/rethnet_eth/src/signature.rs

```diff
@@ -4,62 +4,72 @@
 // For the original context see: https://github.com/gakonst/ethers-rs/blob/cba6f071aedafb766e82e4c2f469ed5e4638337d/ethers-core/src/types/signature.rs
 
 use core::fmt;
+#[cfg(feature = "std")]
 use std::str::FromStr;
 
-use secp256k1::{
-    ecdsa::{RecoverableSignature, RecoveryId},
-    PublicKey, Secp256k1, SecretKey, SignOnly, ThirtyTwoByteHash,
+use k256::{
+    ecdsa::{
+        signature::hazmat::PrehashSigner, RecoveryId, Signature as ECDSASignature, SigningKey,
+        VerifyingKey,
+    },
+    elliptic_curve::sec1::ToEncodedPoint,
+    FieldBytes, PublicKey, SecretKey,
 };
 use sha3::{Digest, Keccak256};
 
 use crate::{utils::hash_message, Address, B256, U256};
 
 /// Converts a [`PublicKey`] to an [`Address`].
 pub fn public_key_to_address(public_key: PublicKey) -> Address {
-    let hash = Keccak256::digest(&public_key.serialize_uncompressed()[1..]);
+    let pk = public_key.to_encoded_point(/* compress = */ false);
+    let hash = Keccak256::digest(&pk.as_bytes()[1..]);
     // Only take the lower 160 bits of the hash
     Address::from_slice(&hash[12..])
 }
 
-/// Converts a private to an address using the provided context.
+/// Converts a secret key in a hex string format to an address.
 ///
 /// # Examples
 ///
 /// ```
-/// use rethnet_eth::signature::private_key_to_address;
-/// use secp256k1::Secp256k1;
+/// use rethnet_eth::signature::secret_key_to_address;
 ///
-/// let context = Secp256k1::signing_only();
-/// let private_key = "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80";
+/// let secret_key = "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80";
 ///
-/// let address = private_key_to_address(&context, private_key).unwrap();
+/// let address = secret_key_to_address(secret_key).unwrap();
 /// ```
-pub fn private_key_to_address(
-    context: &Secp256k1<SignOnly>,
-    private_key: &str,
-) -> Result<Address, secp256k1::Error> {
-    private_to_public_key(context, private_key).map(public_key_to_address)
+pub fn secret_key_to_address(secret_key: &str) -> Result<Address, SignatureError> {
+    let sk = secret_key_from_str(secret_key)?;
+    Ok(public_key_to_address(sk.public_key()))
 }
 
-fn private_to_public_key(
-    context: &Secp256k1<SignOnly>,
-    private_key: &str,
-) -> Result<PublicKey, secp256k1::Error> {
-    let private_key = private_key.strip_prefix("0x").unwrap_or(private_key);
-
-    SecretKey::from_str(private_key).map(|secret_key| secret_key.public_key(context))
+/// Converts a hex string to a secret key.
+pub fn secret_key_from_str(secret_key: &str) -> Result<SecretKey, SignatureError> {
+    let sk = if let Some(stripped) = secret_key.strip_prefix("0x") {
```

Comment:
Ok I thought it'd be obvious since it's of the form `let sk = ... secret_key ...`, but happy to change it. Done in [b746549.](https://github.com/NomicFoundation/hardhat/pull/4465/commits/b7465495b2a5be74c1f2c384985b7b8caae9d279)